python/lsst/analysis/tools/statistics.py

```diff
@@ -19,14 +19,152 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <https://www.gnu.org/licenses/>.
 
-from functools import partial
-
-import scipy.stats as sps
-
 __all__ = (
+    "divide",
+    "flux_to_mag",
+    "nanmax",
+    "nanmean",
+    "nanmin",
+    "nanmedian",
     "nansigmaMad",
+    "nanstd",
     "sigmaMad",
+    "sqrt",
 )
 
-nansigmaMad = partial(sps.median_abs_deviation, scale="normal", nan_policy="omit")
-sigmaMad = partial(sps.median_abs_deviation, scale="normal", nan_policy="propagate")
+import warnings
+from typing import cast
+
+import astropy.units as u
+import numpy as np
+import scipy.stats as sps
+
+from .interfaces import Scalar, Vector
+from .warning_control import (
+    filterwarnings_action,
+    numpy_all_nan,
+    numpy_divide_zero_divide,
+    numpy_divide_zero_log,
+    numpy_divide_zero_log10,
+    numpy_dof_zero,
+    numpy_invalid_value_divide,
+    numpy_invalid_value_log,
+    numpy_invalid_value_log10,
+    numpy_invalid_value_sqrt,
+    numpy_mean_empty,
+)
+
+
+def divide(dividend: Scalar | Vector, divisor: Scalar | Vector) -> Scalar | Vector:
+    """Return dividend/divisor."""
+    with warnings.catch_warnings():
```

Comment:
Can we not catch the issue rather than the warnings? 

if divisor == 0:
    return NaN

Or something? I guess it does the same thing but feels more honest. Also doesn't solve my problem that what I really want is it to also tell me why it has failed. Maybe we just need a metric for each whatever that is number of 0, number of NaN etc.